scala-ts-mode.el

```diff
@@ -549,6 +573,10 @@ Return nil if there is no name or if NODE is not a defun node."
      (treesit-node-text
       (treesit-node-child-by-field-name node "name")
       t))
+    ("val_definition"
```

Comment:
I'm unsure if values are a good target to show in imenu or related packages. I my opinion you might have quite a few vals in scope depending on your style of programming, and that may end up cluttering the imenu entries.

What's your take? I'm fine with adding this in if it's something that would benefit you and others, since it's not hard narrowing down imenu entries

Reply:
This change is not affecting imenu. Value of `treesit-simple-imenu-settings` is not dependent on this (as far as I can see).

This change is for when you run `(treesit-defun-name (treesit-defun-at-point))` on a `val` to get its name.

Reply:
I see what you mean, since we're not really modifying the imenu settings, the vals won't show up. I'm fine with the change, but I wanna ask what is the intended use case for adding it in

Reply:
I want to use it in my [println-debugger](https://github.com/VlachJosef/println-debugger/blob/master/println-debugger-scala.el#L37) package to be able to to get a name of enclosing `val`.  See a README for some example gifs of what that package does.

Basically I want to be able to run `(treesit-defun-name (treesit-defun-at-point))` anywhere in scala file and get back the closest most meaningful identifier, be it `val`, `defun`, `class` etc.

Reply:
That actually looks like a really useful package 👀